spec_integration/facter_spec.rb

```diff
@@ -99,6 +123,27 @@ def tmp_filename(tmp_filename)
     end
 
     context 'with external facts' do
+      context 'with facts that have null bytes' do
+        before do
+          Facter.search_external([ext_facts_dir])
+          data = { "bad_key\u0000" => 'valid value',
+                   'valid_key' => "\x00" }
+          write_to_file(tmp_filename('bad_facts.yaml'), YAML.dump(data))
+        end
+
+        it 'shows errors in the logs when trying to read in the facts with bad keys' do
+          expect { Facter.value("bad_key\u0000") }.to raise_error(ArgumentError)
```

Comment:
When trying to read in fact keys, facter ends up calling `load` in the fact loader, which attempts to read in a file name with the key with the null byte [here](https://github.com/puppetlabs/facter/blob/85797f4/lib/facter/custom_facts/util/loader.rb#L26). The error comes from `File.join`, which raises an Argument error `string contains null byte`. Since the error seemed reasonable, I left it alone for now. I realize now it doesn't contain any indication of *which* string contains a null byte, which could make it harder for debugging later.